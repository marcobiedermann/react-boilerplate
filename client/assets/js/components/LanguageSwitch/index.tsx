import React from 'react';
import i18n from '../../i18n';

function changeLanguage(lng: string): void {
  i18n.changeLanguage(lng);
}

interface LanguageSwitchProps {
  languages: string[];
}

const LanguageSwitch: React.FC<LanguageSwitchProps> = (props) => {
  const { languages } = props;

  return (
    <ul>
      {languages.map((language) => (
        <li key={language}>
          <button onClick={() => changeLanguage(language)}>{language}</button>
        </li>
      ))}
    </ul>
  );
};

export default LanguageSwitch;
