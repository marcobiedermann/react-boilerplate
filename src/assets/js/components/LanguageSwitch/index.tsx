import React from 'react';
import i18n from '../../i18n';

function changeLanguage(lng: string): void {
  i18n.changeLanguage(lng);
}

export interface LanguageSwitchProps {
  languages: string[];
}

function LanguageSwitch(props: LanguageSwitchProps): JSX.Element {
  const { languages } = props;

  return (
    <ul>
      {languages.map((language) => (
        <li key={language}>
          <button onClick={() => changeLanguage(language)}>{language}</button>
        </li>
      ))}
    </ul>
  );
}

export default LanguageSwitch;
