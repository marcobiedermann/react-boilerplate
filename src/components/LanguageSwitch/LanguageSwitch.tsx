import { useTranslation } from 'react-i18next';

interface LanguageSwitchProps {
  languages: string[];
}

function LanguageSwitch(props: LanguageSwitchProps): JSX.Element {
  const { languages } = props;
  const { i18n } = useTranslation();

  async function changeLanguage(lng: string): Promise<void> {
    await i18n.changeLanguage(lng);
  }

  return (
    <ul>
      {languages.map((language) => (
        <li key={language}>
          <button onClick={() => changeLanguage(language)} type="button">
            {language}
          </button>
        </li>
      ))}
    </ul>
  );
}

export type { LanguageSwitchProps };
export default LanguageSwitch;
