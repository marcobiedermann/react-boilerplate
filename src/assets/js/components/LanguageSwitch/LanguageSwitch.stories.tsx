/* eslint-disable import/no-extraneous-dependencies */

import { ComponentMeta, ComponentStoryObj } from '@storybook/react';
import LanguageSwitch from '.';

export default {
  component: LanguageSwitch,
} as ComponentMeta<typeof LanguageSwitch>;

export const Default: ComponentStoryObj<typeof LanguageSwitch> = {
  args: {
    languages: ['de', 'en', 'fr'],
  },
};
