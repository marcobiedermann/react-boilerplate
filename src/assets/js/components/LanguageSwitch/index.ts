import LanguageSwitch from './LanguageSwitch';

export default LanguageSwitch;
export * from './LanguageSwitch';
