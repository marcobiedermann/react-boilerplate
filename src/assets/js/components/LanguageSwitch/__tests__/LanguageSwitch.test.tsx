import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import * as stories from '../LanguageSwitch.stories';

jest.mock('../../../utilities/environment');

const { Default } = composeStories(stories);

describe('language switch', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
