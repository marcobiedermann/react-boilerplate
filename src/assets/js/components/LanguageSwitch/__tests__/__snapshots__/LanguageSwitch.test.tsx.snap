// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language switch should render correctly 1`] = `
<ul>
  <li>
    <button
      type="button"
    >
      de
    </button>
  </li>
  <li>
    <button
      type="button"
    >
      en
    </button>
  </li>
  <li>
    <button
      type="button"
    >
      fr
    </button>
  </li>
</ul>
`;
